test/Operators/hybrid/2d.jl

```diff
@@ -172,63 +188,154 @@ end
+    @test conv_comp_div[1] ≈ 0.5 atol = 0.1
+    @test conv_comp_div[2] ≈ 0.5 atol = 0.1
+    @test conv_comp_div[3] ≈ 0.5 atol = 0.1
 end
 
+@testset "1D SE, 1D FD Extruded Domain horz & vert divergence operator, with Dirichelet BCs" begin
+
+    # Divergence operator in 2D Cartesian domain with
+    # Cₕ = (cₕ, 0), Cᵥ = (0, cᵥ)
+    # ∇ₕ⋅(Cₕ * f) + ∇ᵥ⋅(Cᵥ * f)
+    # here cₕ == cᵥ == 1
+
+    # NOTE: the equation setup is only correct for Cartesian domains!
+
+    n_elems_seq = 2 .^ (6, 7, 8, 9)
+    err, Δh = zeros(length(n_elems_seq)), zeros(length(n_elems_seq))
+
+    for (k, n) in enumerate(n_elems_seq)
+        hv_center_space, hv_face_space =
+            hvspace_2D(xlim = (-1, 1), zlim = (-1, 1), helem = n, velem = n)
+
+        Δh[k] = 1.0 / n
+
+        function divergence(f)
+
+            divuf = Fields.FieldVector(h = zeros(eltype(f), hv_center_space))
+            h = f.h
+            dh = divuf.h
+
+            # vertical advection no inflow at bottom
+            # and outflow at top
+            Ic2f = Operators.InterpolateC2F(
+                top = Operators.SetValue(0.0),
+                bottom = Operators.SetValue(0.0),
+            )
+            divf2c = Operators.DivergenceF2C()
+            # only upward component of divergence
+            @. dh = divf2c(Ic2f(h) * Geometry.WVector(1.0))
+
+            # only horizontal component of divergence
+            hdiv = Operators.Divergence()
+            @. dh += hdiv(h * Geometry.UVector(1.0)) # add the two components for full divergence +=
+            Spaces.weighted_dss!(dh)
+
+            return divuf
+        end
+
+        # A horizontal Gaussian blob, centered at (-x_0, -z_0),
+        # with `a` the height of the blob and σ the standard deviation
+        function h_blob(x_0, z_0, a = 1.0, σ = 0.2)
+            coords = Fields.coordinate_field(hv_center_space)
+            f = map(coords) do coord
+                a * exp(-((coord.x + x_0)^2 + (coord.z + z_0)^2) / (2 * σ^2))
+            end
+
+            return f
+        end
+
+        # Spatial derivative of a horizontal Gaussian blob, centered at (-x_0, -z_0),
+        # with `a` the height of the blob and σ the standard deviation
+        function div_h_blob(x_0, z_0, a = 1.0, σ = 0.2)
+            coords = Fields.coordinate_field(hv_center_space)
+            div_uf = map(coords) do coord
+                -a * (exp(-((coord.x + x_0)^2 + (coord.z + z_0)^2) / (2 * σ^2))) / (σ^2) * ((coord.x + x_0) + (coord.z + z_0))
+            end
+
+            return div_uf
+        end
+
+        f = Fields.FieldVector(h = h_blob(0.0, 0.0))
+
+        divuf = divergence(f)
+        exact_divuf = div_h_blob(0.0, 0.0)
+
+        err[k] = norm(exact_divuf .- divuf.h)
+    end
+    # Divergence convergence rate
+    conv_comp_div = convergence_rate(err, Δh)
+    @test err[3] ≤ err[2] ≤ err[1] ≤ 0.1
+    @test conv_comp_div[1] ≈ 2 atol = 0.1
+    @test conv_comp_div[2] ≈ 2 atol = 0.1
+    @test conv_comp_div[3] ≈ 2 atol = 0.1
```

Comment:
@charleskawczynski 
Here is the Interpolate + Divergence test with Dirichelet SetValue BC that indeed is 2nd order